packages/Main/src/Parser/GeotiffParser.ts

```
@@ -0,0 +1,316 @@
+            this.dataType,
+        );
+
+        texture.flipY = true;
+        texture.needsUpdate = true;
+
+
+        return texture;
+    }
+
+    extentToImageWindow(extent: Extent) {
+        const [oX, oY] = this.origin;
+        const [resX, resY] = this.resolution;
+
+        const wnd = [
+            Math.round((extent.west - oX) / resX),
+            Math.round((extent.north - oY) / resY),
+            Math.round((extent.east - oX) / resX),
+            Math.round((extent.south - oY) / resY),
+        ];
+
+        const xMin = Math.min(wnd[0], wnd[2]);
+        let xMax = Math.max(wnd[0], wnd[2]);
+        const yMin = Math.min(wnd[1], wnd[3]);
+        let yMax = Math.max(wnd[1], wnd[3]);
+
+        // prevent zero-sized requests
+        if ((xMax - xMin) === 0) { xMax += 1; }
+        if ((yMax - yMin) === 0) { yMax += 1; }
+
+        return [xMin, yMin, xMax, yMax];
+    }
+
+    async #readImage(
+        options: {
+            window?: Array<number>,
+            outputWidth: number,
+            outputHeight: number,
+            resampleMethod?: string,
+            pool?: Pool,
+        },
+    ): Promise<TypedArrayWithDimensions> {
+        const {
+            window,
+            outputWidth,
+            outputHeight,
+            resampleMethod = 'nearest',
+            pool,
+        } = options;
+
+        try {
+            return <TypedArrayWithDimensions> await this.image.readRasters({
+                window,
+                pool,
+                width: outputWidth,
+                height: outputHeight,
+                resampleMethod,
+                interleave: true,
+            });
+        // TODO: Do we still need to catch this error?
+        // eslint-disable-next-line @typescript-eslint/no-explicit-any
+        } catch (error: any) {
+            if (error.toString() === 'AggregateError: Request failed') {
+                await new Promise((resolve) => {
+                    setTimeout(resolve, 100);
```

If you want to keep this, I suppose that you should implement some kind of retry counter, what do you think? And maybe log something to avoid silent fail?
And don't you want to check against `error.code` instead of `error.toString()`?
Don't forget to delete `//TODO` comment above

---

I tested again and could not reproduce the error. It occurred when I started working on Geotiff but the code evolution must have neutralized it. I removed this part (the whole method since it was just a wrapper around `GeoTIFFImage.readRasters` that managed the error).